src/Programs/ui/ProgramsRoot.tsx

```diff
@@ -33,8 +33,9 @@ export function ProgramsRoot(): React.ReactElement {
       return (a.create?.level ?? 0) - (b.create?.level ?? 0);
     });
 
+  const programsRef = useRef(programs);
   useEffect(() => {
-    programs.forEach((p) => {
+    programsRef.current.forEach((p) => {
       if (ProgramsSeen.includes(p.name)) return;
       ProgramsSeen.push(p.name);
```

Comment:
I've just taken a closer look at this effect... what is it doing and why?
`ProgramsSeen` is only used inside of this `useEffect` - it does not make any sense to me.
I'll drop it.